quetz/rest_models.py

```diff
@@ -7,12 +7,19 @@
 import uuid
 from datetime import date, datetime
 from enum import Enum
-from typing import Annotated, Dict, Generic, List, Optional, TypeVar, Union
+from typing import Dict, Generic, List, NewType, Optional, TypeVar, Union
 
-from pydantic import BaseModel, ConfigDict, Field, field_validator, model_validator
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    constr,
+    field_validator,
+    model_validator,
+)
 
 T = TypeVar('T')
-URLType = Annotated[str, Field(pattern="^(http|https)://.+")]
+URLType = NewType('URLType', constr(pattern="^(http|https)://.+|None|^(\\[.*\\])+$"))
```

Comment:
Why do you have a URL type and define a pattern? https://docs.pydantic.dev/latest/usage/types/urls/

Reply:
Hm, my thought was that due to implementation of `AnyHttpUrl` in Pydantic
```
AnyHttpUrl = Annotated[Url, UrlConstraints(allowed_schemes=['http', 'https'])]
```
this won't work as well.
But afaik, they only require Python 3.7 which we do as well.